lib/rucio/web/ui/flask/common/utils.py

```diff
@@ -0,0 +1,626 @@
+# limitations under the License.
+#
+# Authors:
+# - Thomas Beermann <thomas.beermann@cern.ch>, 2014-2020
+# - Mario Lassnig <mario.lassnig@cern.ch>, 2014-2018
+# - Vincent Garonne <vincent.garonne@cern.ch>, 2015
+# - Martin Barisits <martin.barisits@cern.ch>, 2016-2020
+# - Ruturaj Gujar <ruturaj.gujar23@gmail.com>, 2019
+# - Jaroslav Guenther <jaroslav.guenther@cern.ch>, 2019-2020
+# - Eli Chadwick <eli.chadwick@stfc.ac.uk>, 2020
+# - Benedikt Ziemons <benedikt.ziemons@cern.ch>, 2020
+# - Patrick Austin <patrick.austin@stfc.ac.uk>, 2020
+#
+# PY3K COMPATIBLE
+
+import re
+import sys
+from json import dumps, load
+from os.path import dirname, join
+from time import time
+
+from flask import request, render_template, redirect, make_response
+
+from rucio.api import authentication as auth, identity
+from rucio.api.account import account_exists, get_account_info, list_account_attributes
+from rucio.common.config import config_get, config_get_bool
+from rucio.core import identity as identity_core, vo as vo_core
+from rucio.db.sqla.constants import AccountType, IdentityType
+
+try:
+    from urllib import unquote, quote
+except ImportError:
+    from urllib.parse import unquote, quote
+
+
+if sys.version_info > (3, 0):
+    long = int
+    unicode = str
+
+escapefunc = None
+try:
+    import html
+    escapefunc = html.escape
+except ImportError:
+    import cgi
+    escapefunc = cgi.escape
+
+try:
+    from onelogin.saml2.auth import OneLogin_Saml2_Auth
+    SAML_SUPPORT = True
+except:
+    SAML_SUPPORT = False
+
+# check if there is preferred server side config for webui authentication
+AUTH_TYPE = config_get('webui', 'auth_type', False, None)
+if AUTH_TYPE == 'oidc':
+    try:
+        AUTH_ISSUER_WEBUI = config_get('webui', 'auth_issuer')
+    except:
+        render_template("problem.html", msg="Please specify auth_issuer in the [webui] section of the Rucio configuration.")
+
+# if no specific config on the server side - we collect information
+# about all authentication options, in particular OIDC
+AUTH_ISSUERS = []
+if not AUTH_TYPE:
+    IDPSECRETS = config_get('oidc', 'idpsecrets', False, None)
+    try:
+        with open(IDPSECRETS) as client_secret_file:
+            client_secrets = load(client_secret_file)
+            for iss in client_secrets:
+                AUTH_ISSUERS.append(iss.upper())
+    except:
+        AUTH_ISSUERS = []
+
+MULTI_VO = config_get_bool('common', 'multi_vo', raise_exception=False, default=False)
+
+# excluded characters for injected JavaScript variables
+VARIABLE_VALUE_REGEX = re.compile(r"^[\w\- /=,.+*#()\[\]]*$", re.UNICODE)
+
+# TO-DO !!! Remove passing data with account and other params to the functions
+# catch these from the webpy input() storage object
+# will allow to remove also lines around each use of select_account_name
+
+COOKIES = []
+
+
+def html_escape(s, quote=True):
+    return escapefunc(s, quote)
+
+
+def prepare_saml_request():
+    """
+    TODO: Validate for Flask
```

Comment:
I guess this should be done before the merge

Reply:
See previous comments.